### DISCUSSION TITLE
need a proper setting of rpath env variable to be set or get rpath not set error

### DISCUSSION THREAD

**Discussion by dvandusen** [Homebrew/discussions#1502] [Everyday usage]
Thanks for working on brew, but I have to raise an issue on brew with zstd v 1.5:

==> Installing dependencies for curl: zstd
==> Installing curl dependency: zstd
==> Pouring zstd-1.5.0.big_sur.bottle.tar.gz
tar: Error opening archive: Failed to open '/Users/davandusen1/Library/Caches/Homebrew/downloads/e427992d8fdd4c16a3cc7e8e4abfeb66fe7df4cf34ac499835f94d0a4ce30806--zstd-1.5.0.big_sur.bottle.tar.gz'
Error: Failure while executing; `tar xof /Users/davandusen1/Library/Caches/Homebrew/downloads/e427992d8fdd4c16a3cc7e8e4abfeb66fe7df4cf34ac499835f94d0a4ce30806--zstd-1.5.0.big_sur.bottle.tar.gz -C /var/folders/jg/gdp0s7n50_932tvxnk42_r040000gn/T/d20210520-39251-m8t56g` exited with 1. Here's the output:
tar: Error opening archive: Failed to open '/Users/davandusen1/Library/Caches/Homebrew/downloads/e427992d8fdd4c16a3cc7e8e4abfeb66fe7df4cf34ac499835f94d0a4ce30806--zstd-1.5.0.big_sur.bottle.tar.gz'

Warning: Bottle installation failed: building from source.
Error: An exception occurred within a child process:
  NameError: undefined local variable or method `rpath' for #<Formulary::FormulaNamespace2a095c4edce22fd6892ec28a460cb294::Zstd:0x00007fd2b41c4b00>
Did you mean?  path
               @path
               paths
               patch
(base) dvandusen17:~ davandusen1$ brew install zstd
==> Downloading https://homebrew.bintray.com/bottles/zstd-1.5.0.big_sur.bottle.tar.gz
##O#- #
curl: (22) The requested URL returned error: 404 Not Found
Error: Failed to download resource "zstd"
Download failed: https://homebrew.bintray.com/bottles/zstd-1.5.0.big_sur.bottle.tar.gz
Warning: Bottle installation failed: building from source.
==> Downloading https://github.com/facebook/zstd/archive/v1.5.0.tar.gz
Already downloaded: /Users/davandusen1/Library/Caches/Homebrew/downloads/04e7b7abaf3f22d0e022f5cbbc2170f9274a70f7f1d1cbdd6bf34f7a8c6981fb--zstd-1.5.0.tar.gz
Error: An exception occurred within a child process:
  NameError: undefined local variable or method `rpath' for #<Formulary::FormulaNamespace2a095c4edce22fd6892ec28a460cb294::Zstd:0x00007ffa8fa458a8>
Did you mean?  path
               @path
               paths
               patch

I don't understand why rpath is even needed or ever used given it is brew. It is holding back all of these updates:

==> Upgrading 40 outdated packages:
neo4j 4.2.5 -> 4.2.6
libheif 1.11.0_2 -> 1.12.0
mysql-client 8.0.23 -> 8.0.25
arpack 3.8.0 -> 3.8.0_1
hdf5 1.12.0_1 -> 1.12.0_4
rustup-init 1.24.1 -> 1.24.2
pango 1.48.4 -> 1.48.5
glib 2.68.1 -> 2.68.2
aom 3.0.0 -> 3.1.0
qt 6.0.3 -> 6.1.0_1
open-mpi 4.1.1 -> 4.1.1_2
innotop 1.12.0_3 -> 1.13.0
bash 5.1.4 -> 5.1.8
openexr 2.5.5 -> 3.0.3
gradle 7.0 -> 7.0.2_1
openblas 0.3.15 -> 0.3.15_1
curl 7.76.1 -> 7.76.1_1
phpunit 9.5.4 -> 9.5.4_1
octave 6.2.0_2 -> 6.2.0_3
scala 2.13.5 -> 2.13.6
mysql 8.0.23_1 -> 8.0.25
graphviz 2.47.0 -> 2.47.1
sbt 1.5.1 -> 1.5.2
libtasn1 4.16.0_1 -> 4.17.0
oniguruma 6.9.6 -> 6.9.7.1
suite-sparse 5.9.0 -> 5.10.0
sip 6.0.3_1 -> 6.1.0
libidn 1.36 -> 1.37
jupyterlab 3.0.14 -> 3.0.16
fltk 1.3.5 -> 1.3.6
node 16.0.0_1 -> 16.2.0
mydumper 0.10.3 -> 0.10.5
fftw 3.3.9 -> 3.3.9_1
qrupdate 1.1.2_13 -> 1.1.2_14
imagemagick 7.0.11-10 -> 7.0.11-13_1
guile 3.0.6 -> 3.0.7
percona-toolkit 3.3.0 -> 3.3.1
subversion 1.14.1 -> 1.14.1_1
libx11 1.7.0 -> 1.7.1
protobuf 3.15.8 -> 3.17.0_1
==> Upgrading curl 7.76.1 -> 7.76.1_1
==> Downloading https://homebrew.bintray.com/bottles/zstd-1.5.0.big_sur.bottle.tar.gz
##O#- #
curl: (22) The requested URL returned error: 404 Not Found
Error: Failed to download resource "zstd"
Download failed: https://homebrew.bintray.com/bottles/zstd-1.5.0.big_sur.bottle.tar.gz
Warning: Bottle installation failed: building from source.
==> Downloading https://github.com/facebook/zstd/archive/v1.5.0.tar.gz
Already downloaded: /Users/davandusen1/Library/Caches/Homebrew/downloads/04e7b7abaf3f22d0e022f5cbbc2170f9274a70f7f1d1cbdd6bf34f7a8c6981fb--zstd-1.5.0.tar.gz
==> Downloading https://homebrew.bintray.com/bottles/curl-7.76.1_1.big_sur.bottle.tar.gz
##O#- #
curl: (22) The requested URL returned error: 404 Not Found
Error: Failed to download resource "curl"
Download failed: https://homebrew.bintray.com/bottles/curl-7.76.1_1.big_sur.bottle.tar.gz
Warning: Bottle installation failed: building from source.
==> Downloading https://homebrew.bintray.com/bottles/zstd-1.5.0.big_sur.bottle.tar.gz
##O#- #
curl: (22) The requested URL returned error: 404 Not Found
Error: Failed to download resource "zstd"
Download failed: https://homebrew.bintray.com/bottles/zstd-1.5.0.big_sur.bottle.tar.gz
Warning: Bottle installation failed: building from source.
==> Downloading https://github.com/facebook/zstd/archive/v1.5.0.tar.gz
Already downloaded: /Users/davandusen1/Library/Caches/Homebrew/downloads/04e7b7abaf3f22d0e022f5cbbc2170f9274a70f7f1d1cbdd6bf34f7a8c6981fb--zstd-1.5.0.tar.gz
==> Downloading https://github.com/curl/curl/commit/8bdde6b14ce3b5fd71c772a578fcbd4b6fa6df19.patch?full_index=1
Already downloaded: /Users/davandusen1/Library/Caches/Homebrew/downloads/c44e60681057e56d5d91d77a1d212207a8061ec3a0a0ca99abca9d2e12e17436--8bdde6b14ce3b5fd71c772a578fcbd4b6fa6df19.patch
==> Downloading https://curl.se/download/curl-7.76.1.tar.bz2
Already downloaded: /Users/davandusen1/Library/Caches/Homebrew/downloads/347969ec9b9438df348361fb69dfa383c76730ef3c111ecce598ea7e728af899--curl-7.76.1.tar.bz2
==> Installing dependencies for curl: zstd

So, as I can understand, because zstd is not available it has to be built, and because rpath is not set there is no upgrading being done.

**Comment by carlocab** [comment]
You likely have an outdated version of `brew`. What's the output of `brew config` and `brew doctor`?

**Comment by dvandusen** [comment]
Carlo: This morning I went thru a long but productive debug session.  I 
'updated' 4 times at least, and 'cleaned' several times.  Doctor was 
able to show what broke each time and after all of that 'upgrade' 
finally worked. I also went the 'curl' procedure for hstd in the middle 
of the above, and suspect that forcing hstd to install (at v 1.5) 
changed some of the dependencies just enough to allow the last 'update' 
to work.  Since I lost track of exactly what order this all happened 
in, I cannot really pinpoint the issue's cause.

I must tell you and other devs on brew that I truly appreciate your 
efforts.  Please feel great about it.  I have been in the business a 
very long time.

Thx

D
